Amazon.IonObjectMapper.Test/TestObjects.cs

```diff
@@ -483,4 +483,351 @@ public static string PrettyString(IDictionary<string, int> dictionary)
+        }
+
+        public string GetProtectedProperty()
+        {
+            return $"{ProtectedProperty}";
+        }
+
+        public string GetDefaultProperty()
+        {
+            return $"{DefaultProperty}";
+        }
+    }
+
+    public class ClassWithReadonlyProperties
+    {
+        public ClassWithReadonlyProperties() { }
+
+        public ClassWithReadonlyProperties(string PublicProperty, string DefaultProperty, string ProtectedProperty, string ProtectedInternalProperty,
+            string InternalProperty, string PrivateProperty, string PrivateProtectedProperty)
+        {
+            this.PublicProperty = PublicProperty;
+            this.DefaultProperty = DefaultProperty;
+            this.ProtectedProperty = ProtectedProperty;
+            this.ProtectedInternalProperty = ProtectedInternalProperty;
+            this.InternalProperty = InternalProperty;
+            this.PrivateProperty = PrivateProperty;
+            this.PrivateProtectedProperty = PrivateProtectedProperty;
+        }
+
+        public string PublicProperty
+        {
+            get;
+        }
+
+        string DefaultProperty
+        {
+            get;
+        }
+
+        protected string ProtectedProperty
+        {
+            get;
+        }
+
+        protected internal string ProtectedInternalProperty
+        {
+            get;
+        }
+
+        internal string InternalProperty
+        {
+            get;
+        }
+
+        private string PrivateProperty
+        {
+            get;
+        }
+
+        private protected string PrivateProtectedProperty
+        {
+            get;
+        }
+
+        public string GetPrivateProtectedProperty()
+        {
+            return $"{PrivateProtectedProperty}";
+        }
+
+        public string GetPrivateProperty()
+        {
+            return $"{PrivateProperty}";
+        }
+
+        public string GetProtectedProperty()
+        {
+            return $"{ProtectedProperty}";
+        }
+
+        public string GetDefaultProperty()
+        {
+            return $"{DefaultProperty}";
+        }
+    }
+
+
+    public class ClassWithIonPropertyNamesAttribute
+    {
+        public ClassWithIonPropertyNamesAttribute() { }
+
+        public ClassWithIonPropertyNamesAttribute(string PublicProperty, string DefaultProperty, string ProtectedProperty, string ProtectedInternalProperty,
+            string InternalProperty, string PrivateProperty, string PrivateProtectedProperty)
+        {
+            this.PublicProperty = PublicProperty;
+            this.DefaultProperty = DefaultProperty;
+            this.ProtectedProperty = ProtectedProperty;
+            this.ProtectedInternalProperty = ProtectedInternalProperty;
+            this.InternalProperty = InternalProperty;
+            this.PrivateProperty = PrivateProperty;
+            this.PrivateProtectedProperty = PrivateProtectedProperty;
+        }
+
+        [IonPropertyName("PublicPropertyz")]
```

Comment:
Is the 'z' appended at the end just to avoid a naming conflict? If so, how about just "Public Property" instead? (Likewise for the other `[IonPropertyName]`s)

Reply:
Updated